akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -759,7 +798,7 @@ private[akka] class ShardRegion(
           after.map(_.address).getOrElse(""))
       coordinator = None
       startRegistration()
-    }
+    } else reRegisterIfCoordinatorNotUp()
```

Comment:
do we have to do this for all membership changes, or would it be enough when
```
if (coordinator.nonEmpty && before.map(_.status) != after.map(_.status))
```

Reply:
Since the coordinator isn't strictly guaranteed to be on before/after.  I guess for membership changes it could compare previous coordinator node status with new coordinator node status....